sdk/core/core-http/lib/util/atomXmlHelper.ts

```diff
@@ -0,0 +1,338 @@
+  });
+
+  if (contentElementNames && contentElementNames[0]) {
+    const contentRootElementName = contentElementNames[0];
+    delete entry.content[contentRootElementName][Constants.XML_METADATA_MARKER];
+    result = entry.content[contentRootElementName];
+
+    if (result) {
+      if (entry[Constants.XML_METADATA_MARKER]) {
+        result[Constants.ATOM_METADATA_MARKER] = entry[Constants.XML_METADATA_MARKER];
+      } else {
+        result[Constants.ATOM_METADATA_MARKER] = {};
+      }
+
+      result[Constants.ATOM_METADATA_MARKER]["ContentRootElement"] = contentRootElementName;
+
+      Object.keys(entry).forEach((property: string) => {
+        if (property !== "content" && property !== Constants.XML_METADATA_MARKER) {
+          result[Constants.ATOM_METADATA_MARKER][property] = entry[property];
+        }
+      });
+
+      return result;
+    }
+  }
+
+  return undefined;
+}
+
+/**
+ * @ignore
+ * Utility to help parse given `feed` result
+ * @param feed
+ */
+function parseFeedResult(feed: any): XMLResponseInJSON[] {
+  const result = [];
+  if (typeof feed === "object" && feed != null && feed.entry) {
+    if (Array.isArray(feed.entry)) {
+      feed.entry.forEach((entry: any) => {
+        const parsedEntryResult = parseEntryResult(entry);
+        if (parsedEntryResult) {
+          result.push(parsedEntryResult);
+        }
+      });
+    } else {
+      const parsedEntryResult = parseEntryResult(feed.entry);
+      if (parsedEntryResult) {
+        result.push(parsedEntryResult);
+      }
+    }
+  }
+  return result;
+}
+
+/**
+ * Transforms response to contain the parsed data.
+ * @param nameProperties The set of 'name' properties to be constructed on the
+ * resultant object e.g., QueueName, TopicName, etc.
+ * @param response
+ * @param shouldParseResponse
+ */
+export async function deserializeAtomXmlResponse(
+  nameProperties: string[],
+  response: HttpOperationResponse,
+  shouldParseResponse: boolean
+): Promise<HttpOperationResponse> {
+  let atomResponseInJson: any;
+  try {
+    if (response.bodyAsText && response.bodyAsText.toString().length > 0) {
+      atomResponseInJson = await deserializeAtomXmlToJson(response.bodyAsText);
+      response.parsedBody = atomResponseInJson;
+    }
+  } catch (e) {
+    response.errorBody = {
+      error: { code: "ResponseNotInAtomXMLFormat" }
+    };
+  }
+
+  if (response.status < 200 || response.status > 300) {
+    if (response.errorBody == undefined) {
+      const HttpResponseCodes = Constants.HttpResponseCodes;
+      const statusCode = response.status;
+      if (!isKnownResponseCode(statusCode)) {
+        response.errorBody = {
+          error: { code: `UnrecognizedHttpResponseStatus: ${statusCode}` }
+        };
+      } else {
+        response.errorBody = { error: { code: HttpResponseCodes[statusCode] } };
+      }
+    }
+  }
+
+  // Transform the errorBody to a normalized one
+  const normalizedError = normalizeError(response.errorBody, response);
```

Comment:
Since the first argument is part of the second argument, why not pass just the second argument?

Reply:
Since we are still in process of clarifying purpose of `response.errorBody`, I'd keep this simplification to the end.